server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -454,7 +459,8 @@ public ClusterState execute(ClusterState currentState) {
                             if (metadata.customs() != null) {
                                 for (ObjectObjectCursor<String, Metadata.Custom> cursor : metadata.customs()) {
                                     if (RepositoriesMetadata.TYPE.equals(cursor.key) == false
-                                            && DataStreamMetadata.TYPE.equals(cursor.key) == false) {
+                                            && DataStreamMetadata.TYPE.equals(cursor.key) == false
+                                            && (operatorPrivilegesEnabled == false || "autoscaling".equals(cursor.key) == false)) {
```

Comment:
I wonder if we should just skip the `operatorPrivilegesEnabled` check. It is fine to not restore autoscaling policies for now and we can revisit once we have the proper framework around operator privileges and restore in general.

Reply:
Removing the check makes both the logic and code dependency simpler. Another practical advantage is that Cloud probably will not enable opereator privileges straight away after 7.11 release. Therefore removing this check makes sure the restore behavioiur of autoscaling policy stay the same before and after operator privileges get enabled in Cloud. Having a consistent behaviour is one of the motivations for this work to be included in 7.11. So it seems to make sense. But it would be different than what we have agreed during the meeting. @colings86 Could you please verify whether removing this check is desirable?

Reply:
> But it would be different than what we have agreed during the meeting

I believe we discussed that approach at the meeting and I came out the meeting with the impression that this was the direction we would take (to just disable restore for autoscaling).

Reply:
I'm fine with changing this so we never restore autoscaling policies regardless of whether operator privileges are enabled.

Reply:
Thanks both. I updated the code to drop the operator privileges check.